### PR TITLE
Backport: Update toolkit's gonum to v0.15.0

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

Backport of https://github.com/microsoft/azurelinux/pull/9784 and https://github.com/microsoft/azurelinux/pull/9943

###### Change Log  

- Backport #9784 and #9943

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/52045232

###### Test Methodology

- `sudo  make image CONFIG_FILE=./imageconfigs/core-efi.json QUICK_REBUILD_PACKAGES=y` with `2.0-stable manifests`.
- ~https://dev.azure.com/mariner-org/mariner/_build/results?buildId=613810&view=results~ https://dev.azure.com/mariner-org/mariner/_build/results?buildId=614184&view=results

### CODE REVIEW

File: toolkit/resources/manifests/package/pkggen_core_aarch64.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 filesystem-1.1-20.cm2.aarch64.rpm
-kernel-headers-5.15.162.2-1.cm2.noarch.rpm
+kernel-headers-5.15.160.1-1.cm2.noarch.rpm
```
Comment (mfrw) [inline toolkit/resources/manifests/package/pkggen_core_aarch64.txt:2]: I think this maybe a typo .. changing the manifests.

Comment (dmcilvaney) [reply toolkit/resources/manifests/package/pkggen_core_aarch64.txt:2]: Trying to figure out why, I ran two cherry-pick commands on the two PRs... But yes this is unintended.

Comment (dmcilvaney) [reply toolkit/resources/manifests/package/pkggen_core_aarch64.txt]: Ok, fixed the commit

**Review by mfrw — APPROVED** [review]

File: .github/workflows/go-test-coverage.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ on:
 permissions: read-all
 
 env:
-  EXPECTED_GO_VERSION: "1.20"
+  EXPECTED_GO_VERSION: "1.21"
```
Comment (jslobodzian) [inline .github/workflows/go-test-coverage.yml:15]: Do we need to bump this?

**Review by jslobodzian — APPROVED** [review]

File: .github/workflows/go-test-coverage.yml (hunk 2 above)
Comment (dmcilvaney) [reply .github/workflows/go-test-coverage.yml:15]: Unfortunately yes, one of the libraries has a hard requirement on it.

**Pull request merged**